### ISSUE TITLE
Search matches fields it is not supposed to

### ISSUE THREAD

**Issue by AllKind** [strawberrymusicplayer/strawberry#1290]
- [ x] I have checked the [FAQ](https://wiki.strawberrymusicplayer.org/wiki/FAQ) for answers.
- [ x] I have checked the [Changelog](https://github.com/strawberrymusicplayer/strawberry/blob/master/Changelog) that the issue is not already fixed. 
- [ x] I believe this issue is a bug, and not a general technical issue, question or feature requests that can be discussed on the [forum](https://forum.strawberrymusicplayer.org/). 

**Describe the bug**
If I search for example `title:searchstring`, it will return matches matching the album fields - see screenshot.
I attach another screenshot, where I search for `artist` and it returns results matching the field `title` and/or `album`.
The sorting is set by Genre/Artist/Album in the provided screenshots.

**To Reproduce**
As said above.

**Expected behavior**
The search should be limited to the field which is provided.

**Screenshots:**
![strawberry_search_error2](https://github.com/strawberrymusicplayer/strawberry/assets/3242311/2486be43-4e48-43fe-bd41-b242b0a5e2a4)

---

![strawberry_search_error](https://github.com/strawberrymusicplayer/strawberry/assets/3242311/ed7cc0c4-c954-4815-9f8f-c4f0341ae87f)

**System Information:**
 - Operating system: Linux Mint 21.2 Cinnamon (Custom kernel 5.15.134)
 - Strawberry Version: 1.0.20

**Comment by AllKind** [comment]
Just now I have that search result, I can't even explain. It just seems wrong, as none of the search criteria match:

![strawberry_search_error3](https://github.com/strawberrymusicplayer/strawberry/assets/3242311/8766aad9-667b-4d5e-afdd-ed017e750ada)

**Comment by AllKind** [comment]
Ah, I see! Thank you for answering! And in general for your work!
I recently switched from clementine and if I correctly recall, the space didn't matter.
I wonder if it's worth asking as a feature request to remove the blank(s) before the first work of the search term?
Anyhow I'll close this, as the bug is on me.
Have a good day!

**Comment by jonaski** [comment]
Working on rewriting the entire search, so we might improve this too.

**Comment by AllKind** [comment]
As it just comes to mind:
One thing I was always missing in clementine and I think it's the same for strawberry is a way to define a search term as "strict" or in other words "exact".
For example, as I have a lot of latin tracks a search for very general short words like "title:soy yo" brings up hundreds of matches. It would be nice for me to let strawberry just for titles exactly named "soy yo".

**Comment by jonaski** [comment]
I tested Clementine and it works even with whitespaces after colon like you said, so I became curious and looked into why, and I see it's because it's written quite differently, it just appends everything you type to the search query, while strawberry specifically expects a token after the colon, and ignores it otherwise. It didn't even remember this, but I remember it now.
So I made a fix now so it strips off whitespaces after colon, this way it works with whitespaces in between too.
